src/server/actions/user-actions.test.ts

```diff
@@ -0,0 +1,309 @@
+import { describe, it, expect, vi, afterEach } from 'vitest'
+import { db } from '@/database'
+import { getUserIdByClerkId, getMemberUserPublicKey, findOrCreateSiUserId, createUserAction } from './user-actions'
+
+// Mock the database
+vi.mock('@/database', () => ({
+    db: {
+        selectFrom: vi.fn(),
+        insertInto: vi.fn(),
+        transaction: vi.fn(),
+    },
+}))
+
+describe('User Actions', () => {
+    afterEach(() => {
+        vi.clearAllMocks() // Reset all mocks between tests
```

Comment:
I wonder if we want to add this to the global vitest.setup.ts file in `afterEach`? Is this behavior we'd want for every test? cc @nathanstitt

Reply:
we shouldn't need the `clearAllMocks` because we have `mockReset: true,` turned on in the vitest.config.ts which does this automatically